File: seed/seed.js
import mysql from 'mysql2/promise';
import bcrypt from 'bcryptjs';
import dotenv from 'dotenv';

dotenv.config();

async function seed() {
  const connection = await mysql.createConnection({
    host: process.env.DB_HOST,
    user: process.env.DB_USER,
    password: process.env.DB_PASSWORD,
    database: process.env.DB_NAME,
  });

  try {
    console.log('Seeding database...');

    // Create demo users
    const hashedPassword1 = await bcrypt.hash('password123', 10);
    const hashedPassword2 = await bcrypt.hash('password123', 10);

    await connection.execute(
      'INSERT IGNORE INTO users (email, password, name) VALUES (?, ?, ?)',
      ['alice@example.com', hashedPassword1, 'Alice Johnson']
    );

    await connection.execute(
      'INSERT IGNORE INTO users (email, password, name) VALUES (?, ?, ?)',
      ['bob@example.com', hashedPassword2, 'Bob Smith']
    );

    // Get user IDs
    const [aliceRows] = await connection.execute(
      'SELECT id FROM users WHERE email = ?',
      ['alice@example.com']
    );
    const [bobRows] = await connection.execute(
      'SELECT id FROM users WHERE email = ?',
      ['bob@example.com']
    );

    const aliceId = aliceRows[0].id;
    const bobId = bobRows[0].id;

    // Create sample notes for Alice
    const aliceNotes = [
      { title: 'Project Planning', content: 'Outline for Q2 roadmap...', status: 'active' },
      { title: 'Meeting Notes', content: 'Discussed sprint planning...', status: 'active' },
      { title: 'Ideas', content: 'UI improvements to explore...', status: 'active' },
      { title: 'Learning Resources', content: 'React and TypeScript guides...', status: 'archived' },
      { title: 'Personal Goals', content: 'Career development for 2024...', status: 'active' },
    ];

    for (const note of aliceNotes) {
      await connection.execute(
        'INSERT INTO notes (userId, title, content, status) VALUES (?, ?, ?, ?)',
        [aliceId, note.title, note.content, note.status]
      );
    }

    // Create sample notes for Bob
    const bobNotes = [
      { title: 'Database Schema', content: 'Current schema design...', status: 'active' },
      { title: 'API Documentation', content: 'Endpoints to document...', status: 'active' },
      { title: 'Deployment Checklist', content: 'Pre-deployment tasks...', status: 'archived' },
    ];

    for (const note of bobNotes) {
      await connection.execute(
        'INSERT INTO notes (userId, title, content, status) VALUES (?, ?, ?, ?)',
        [bobId, note.title, note.content, note.status]
      );
    }

    console.log('✅ Database seeded successfully!');
    console.log('Demo users:');
    console.log('- alice@example.com (password: password123)');
    console.log('- bob@example.com (password: password123)');
  } catch (error) {
    console.error('❌ Seeding failed:', error);
  } finally {
    await connection.end();
  }
}

seed();

File: backend/seed.py
"""Database seeding script."""
from app import create_app
from app.extensions import db
from app.models.user import User
from app.models.note import Note
from app.config import DevelopmentConfig


def seed_database():
    """Seed database with demo users and notes."""
    app = create_app(DevelopmentConfig)
    
    with app.app_context():
        db.drop_all()
        db.create_all()

        print('🌱 Seeding database...')

        demo_user = User(
            username='demo_user',
            email='demo@example.com'
        )
        demo_user.set_password('demopass123')

        test_user = User(
            username='test_user',
            email='test@example.com'
        )
        test_user.set_password('testpass123')

        db.session.add(demo_user)
        db.session.add(test_user)
        db.session.commit()

        print(f'✅ Created users: demo_user, test_user')

        demo_notes = [
            Note(
                user_id=demo_user.id,
                title='Project Planning',
                content='Outline for Q2 roadmap and feature prioritization.',
                status='active'
            ),
            Note(
                user_id=demo_user.id,
                title='Meeting Notes',
                content='Discussed sprint planning and team capacity.',
                status='active'
            ),
            Note(
                user_id=demo_user.id,
                title='UI Improvements',
                content='Ideas for improving user interface and experience.',
                status='active'
            ),
            Note(
                user_id=demo_user.id,
                title='Learning Resources',
                content='React and TypeScript guides for team development.',
                status='archived'
            ),
            Note(
                user_id=demo_user.id,
                title='Personal Goals',
                content='Career development objectives for 2024.',
                status='active'
            ),
        ]

        for note in demo_notes:
            db.session.add(note)

        test_notes = [
            Note(
                user_id=test_user.id,
                title='Database Schema',
                content='Current database schema design and optimization.',
                status='active'
            ),
            Note(
                user_id=test_user.id,
                title='API Documentation',
                content='Endpoints to document and API reference guide.',
                status='active'
            ),
            Note(
                user_id=test_user.id,
                title='Deployment Checklist',
                content='Pre-deployment tasks and verification steps.',
                status='archived'
            ),
        ]

        for note in test_notes:
            db.session.add(note)

        db.session.commit()

        print(f'✅ Created {len(demo_notes)} notes for demo_user')
        print(f'✅ Created {len(test_notes)} notes for test_user')
        print('\n📝 Demo Credentials:')
        print('   demo@example.com / demopass123')
        print('   test@example.com / testpass123')
        print('\n✨ Database seeding complete!')


if __name__ == '__main__':
    seed_database()